src/stores/NotificationsStore.js

```diff
@@ -0,0 +1,271 @@
+// @flow
+
+import type { AchievementType, EntityID } from 'brewskey.js-api';
+import type { ObservableMap } from 'mobx';
+import type { Navigation } from '../types';
+
+import { AppState, AsyncStorage } from 'react-native';
+import {
+  action,
+  computed,
+  observable,
+  reaction,
+  runInAction,
+  when,
+} from 'mobx';
+import nullthrows from 'nullthrows';
+import FCM, { FCMEvent } from 'react-native-fcm';
+import DeviceInfo from 'react-native-device-info';
+import AuthStore from './AuthStore';
+import { AchievementStore, KegStore } from './DAOStores';
+import { fetchJSON } from './ApiRequestStores/makeRequestApiStore';
+import CONFIG from '../config';
+
+const BASE_PUSH_URL = `${CONFIG.HOST}api/v2/push`;
+
+const NOTIFICATIONS_STORAGE_KEY = 'notifications';
+
+export type BaseNotification<TNotificationExtraProps: {}> = {
+  ...TNotificationExtraProps,
+  body: string,
+  date: Date,
+  id: string,
+  isRead: string,
+  title: string,
+};
+
+export type LowKegLevelNotificationExtraProps = {
+  beverageId: EntityID,
+  beverageName: string,
+  tapId: EntityID,
+  type: 'lowKegLevel',
+};
+
+export type NewAchievementNotificationExtraProps = {
+  achievementType: AchievementType,
+  type: 'newAchievement',
+};
+
+export type NewFriendRequestNotificationExtraProps = {
+  friendId: EntityID,
+  type: 'newFriendRequest',
+  friendUserName: string,
+};
+
+export type LowKegLevelNotification = BaseNotification<
+  LowKegLevelNotificationExtraProps,
+>;
+
+export type NewAchievementNotification = BaseNotification<
+  NewAchievementNotificationExtraProps,
+>;
+
+export type NewFriendRequestNotification = BaseNotification<
+  NewFriendRequestNotificationExtraProps,
+>;
+
+export type Notification =
+  | LowKegLevelNotification
+  | NewAchievementNotification
+  | NewFriendRequestNotification;
+
+class NotificationsStore {
```

Comment:
Yeah..I know..this store smells not very well. :shy:

Reply:
I actually think this is fine. Push notifications are hacky by nature. What you have here seems reasonable.